src/sentry/db/models/fields/hybrid_cloud_foreign_key.py

```diff
@@ -0,0 +1,59 @@
+"""
+A 'foreign key' which is not enforced in the local database, but triggers eventually consistent delete work in the
+presence of RegionTombstone ControlTombstone model objects through the tasks/deletion.py script.
+
+This column is, in fact, an integer, and cannot enforce integrity directly.  It's main purpose to simply to track
+the need for cleanup in the case of a resource which is tracked in an opposing SiloMode.
+"""
+from __future__ import annotations
+
+from enum import IntEnum
+
+from django.db import models
+
+__all__ = "HybridCloudForeignKey"
+
+from typing import Any, Tuple
+
+from django.apps import apps
+
+
+class HybridCloudForeignKeyCascadeBehavior(IntEnum):
+    CASCADE = 1
+    SET_NULL = 2
+
+
+class HybridCloudForeignKey(models.BigIntegerField):  # type: ignore
+    on_delete: HybridCloudForeignKeyCascadeBehavior
```

Comment:
Possibly, though I did prefer forcing explicitness, especially because this behavior is not captured in migrations the way actual cascades are.  If someone changes the default, there isn't any help warning you that it results in a change in other usage sites, the way a normal foreign key is.